recipes/openimageio/3.x/conanfile.py

```diff
@@ -0,0 +1,303 @@
+        "with_libpng": True,
+        "with_libwebp": True,
+        "with_openjpeg": True,
+        "with_openvdb": False,  # FIXME: broken on M1
+        "with_opencv": False,
+        "with_ptex": True,
+        "with_raw": False,  # libraw is available under CDDL-1.0 or LGPL-2.1, for this reason it is disabled by default
+        "with_tbb": False,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        # Required libraries
+        self.requires("zlib/[>=1.2.11 <2]")
+        self.requires("libtiff/4.7.0")
+        self.requires("imath/3.1.9", transitive_headers=True)
+        self.requires("openexr/3.2.3")
+        if self.options.with_libjpeg == "libjpeg":
+            self.requires("libjpeg/9e")
+        elif self.options.with_libjpeg == "libjpeg-turbo":
+            self.requires("libjpeg-turbo/3.0.4")
+        self.requires("pugixml/1.14")
+        self.requires("libsquish/1.15")
+        self.requires("tsl-robin-map/1.3.0")
+        self.requires("fmt/10.2.1", transitive_headers=True)
+        self.requires("opencolorio/2.4.0")
+
+        # Optional libraries
+        if self.options.with_libjxl:
+            self.requires("libjxl/0.10.3")
+        if self.options.with_libpng:
+            self.requires("libpng/[>=1.6 <2]")
+        if self.options.with_freetype:
+            self.requires("freetype/2.13.2")
+        if self.options.with_hdf5:
+            self.requires("hdf5/1.14.5")
+        if self.options.with_opencv:
+            self.requires("opencv/4.10.0")
+        if self.options.with_tbb:
+            self.requires("onetbb/2021.12.0")
+        if self.options.with_dicom:
+            self.requires("dcmtk/3.6.8")
+        if self.options.with_ffmpeg:
+            self.requires("ffmpeg/7.0.1")
+        # TODO: Field3D dependency
+        if self.options.with_giflib:
+            self.requires("giflib/5.2.2")
+        if self.options.with_libheif:
+            self.requires("libheif/1.18.2")
+        if self.options.with_raw:
+            self.requires("libraw/0.21.3")
+        if self.options.with_openjpeg:
+            self.requires("openjpeg/2.5.2")
+        if self.options.with_openvdb:
+            self.requires("openvdb/11.0.0")
+        if self.options.with_ptex:
+            self.requires("ptex/2.4.2")
+        if self.options.with_libwebp:
+            self.requires("libwebp/1.3.2")
+
+        # TODO: R3DSDK dependency
+        # TODO: Nuke dependency
+
+    def build_requirements(self):
+        # A minimum cmake version is now required that is reasonably new
+        self.build_requires("cmake/[>=3.18.2]")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, 17)
```

Comment:
```suggestion
        check_min_cppstd(self, 17)
```
It's meant to always be set in Conan 2